### PR TITLE
Add IFileInfo overloads for SendFileAsync.

### PR DESCRIPTION
**Pull request by Tratcher**
#527 @davidfowl @lodejard @khellang

### CODE REVIEW

File: src/Microsoft.AspNetCore.Http.Extensions/SendFileResponseExtensions.cs [hunk 1]
```diff
@@ -19,9 +20,66 @@ public static class SendFileResponseExtensions
...
+            {
+                throw new ArgumentNullException(nameof(file));
+            }
+            CheckRange(offset, count, file.Length);
+
+            if (string.IsNullOrEmpty(file.PhysicalPath))
+            {
+                using (var fileContent = file.CreateReadStream())
+                {
+                    fileContent.Seek(offset, SeekOrigin.Begin);
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Http.Extensions/SendFileResponseExtensions.cs]: Is the stream always seekable?

**Comment by davidfowl** [comment]
Small comment around seek-ability of the stream. We should probably throw if we can't seek and the index isn't 0.

**Comment by lodejard** [comment]
and don't seek if the offset is 0

**Comment by Tratcher** [comment]
:up::date: Seek will throw if it's not seekable. No need to add our own exception unless it becomes a frequent point of confusion.

**Pull request merged**